### PR TITLE
Updating Management Track 2 project template

### PR DESCRIPTION
**Pull request by allenjzhang**
The dotnet template for Track 2 management SDK was out of date due to namespace change. Fixed:
1. correct token replacement
2. moved Azure.Core reference to shared Directory.Build.Data.Props from each RP folder's Directory.Build.Props
3.  fixed tab to space in .sln file
4. Added default warning override in the csproj

[AB#3983](https://dev.azure.com/azure-mgmt-ex/4b92ce2a-39c0-4c42-816c-214829fe2a54/_workitems/edit/3983)

### CODE REVIEW

File: eng/templates/Azure.ResourceManager.Template/src/Azure.ResourceManager.Template.csproj [hunk 1]
```diff
@@ -5,4 +5,9 @@
     <Description>Azure management client SDK for Azure resource provider ProviderFullName</Description>
     <PackageTags>azure;management;ProviderNameLowercase</PackageTags>
   </PropertyGroup>
+
+  <PropertyGroup>
+    <!-- TODO: Temp workaround. Should be fixed prior to release  -->
```
Comment (weshaggard) [inline eng/templates/Azure.ResourceManager.Template/src/Azure.ResourceManager.Template.csproj:10]: Should this be in the template? Is there an issue tracking the fixing of the problems? If so please add a link here.

File: eng/templates/Azure.ResourceManager.Template/src/Azure.ResourceManager.Template.csproj [hunk 2]
```diff
@@ -5,4 +5,9 @@
     <Description>Azure management client SDK for Azure resource provider ProviderFullName</Description>
     <PackageTags>azure;management;ProviderNameLowercase</PackageTags>
```
Comment (weshaggard) [inline eng/templates/Azure.ResourceManager.Template/src/Azure.ResourceManager.Template.csproj:6]: Should we also add "resourcemanager" as a tag now?

**Review by weshaggard — APPROVED** [review]
A couple comments but otherwise looks good.

File: eng/templates/Azure.ResourceManager.Template/src/Azure.ResourceManager.Template.csproj (hunk 1 above)
Comment (allenjzhang) [reply eng/templates/Azure.ResourceManager.Template/src/Azure.ResourceManager.Template.csproj]: The AZC0001 is a tricky one as enabling the approved RP namespace can only be done after the initial check-in and successful build. We can only use release process to change afterward.  For AZC0008 should be disabled at infra level.

**Review by heaths — APPROVED** [review]

File: eng/Directory.Build.Data.props [hunk 3]
```diff
@@ -61,6 +61,7 @@
       $(NoWarn);
       CA1812;  <!-- Avoid uninstantiated internal classes. (Result from below including the Azure.Core source) -->
       AZC0012; <!-- Single word class names -->
+      AZC0008;
```
Comment (AlexGhiondea) [inline eng/Directory.Build.Data.props:64]: Could you add a comment like above about what this one disables?

File: eng/templates/Azure.ResourceManager.Template/src/Azure.ResourceManager.Template.csproj [hunk 4]
```diff
@@ -2,7 +2,12 @@
...
     <PackageId>Azure.ResourceManager.Template</PackageId>
-    <Description>Azure management client SDK for Azure resource provider ProviderFullName</Description>
-    <PackageTags>azure;management;ProviderNameLowercase</PackageTags>
+    <Description>Azure Resource Manager client SDK for Azure resource provider ProviderFullName</Description>
+    <PackageTags>azure;management;arm;resource manager;ProviderNameLowercase</PackageTags>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <!-- TODO: As part of release process, this PropertyGroup can be removed once RP is approved for namespace -->
+    <NoWarn>$(NoWarn);AZC0001</NoWarn>
```
Comment (AlexGhiondea) [inline eng/templates/Azure.ResourceManager.Template/src/Azure.ResourceManager.Template.csproj:11]: Could you add a comment describing this warning?

**Review by AlexGhiondea — APPROVED** [review]
Left a couple of comments, but LGTM otherwise.

**Pull request merged**